extensions/content-health-monitor/content_health_utils.py

```diff
@@ -0,0 +1,416 @@
+
+    content_name = content.get("title", "")
+    # Title is optional, if not set use name
+    if not content_name:
+        content_name = content.get("name", "")
+    
+    # Ensure we have a valid URL
+    dashboard_url = content.get("dashboard_url", "")
+    content_url = content.get("content_url", "")
+
+    # Initialize default owner information
+    owner_email = ""
+    owner_full_name = ""
+    
+    # Headers for Connect API
+    headers = {
+        "Authorization": f"Key {api_key}",
+        # Set a custom user agent to enable filtering of activity in Connect instrumentation data
+        "User-Agent": "ContentHealthMonitor/1.0",
+    }
+    
+    # Get additional data if we don't have an error when attempting to get the content details
+    if not str(content_name).startswith("ERROR:"):
+        # Get content owner details
+        try:
+            owner_guid = content.get("owner_guid")
+            if owner_guid:
+                # Get owner details
+                owner = get_user(client, owner_guid)
+                owner_email = owner.get("email", "")
+                owner_first_name = owner.get("first_name", "")
+                owner_last_name = owner.get("last_name", "")
+                owner_full_name = f"{owner_first_name} {owner_last_name}".strip()
+                if not owner_full_name:  # Handle case where both names are empty
+                    owner_full_name = "Unknown"
+        except Exception as e:
+            # If there's an error getting the owner, keep the defaults
+            print(f"Warning: Could not retrieve owner for {guid}: {str(e)}")
+    
+    # Compose URL to logs if we have a dashboard URL, only owner/editor have access
+    if dashboard_url and content.get("app_role") != "viewer":
+        logs_url = f"{dashboard_url}/logs"
+    else:
+        logs_url = ""
+
+    try:
+        # Use the content_url if available, otherwise construct one so we get a proper status code
+        if content_url:
+            content_url = content_url
+        else:
+            # Likely hit error condition in get_content() thus no content_url available
+            content_url = f"{connect_server}/content/{guid}"
+        content_response = requests.get(
+            content_url, 
+            headers=headers,
+            timeout=60, # Max time to wait for a response from the content
+            allow_redirects=True  # Enabled by default in Python requests, included for clarity
+        )
+        
+        # EXTENSION POINT: You can add additional validation beyond HTTP status
+        
+        # Example: Check if response body contains expected text from environment variable
+        # expected_text = os.environ.get("EXPECTED_CONTENT_TEXT")
+        # contains_expected_text = True  # Default to True if no expected text is configured
+        # if expected_text:  # Only perform the check if expected text is configured
+        #     contains_expected_text = expected_text in content_response.text
```

Comment:
I originally had a comment here saying basically, "Love this, maybe you should call it out in the README" and then found that you already do. Awesome!